includes/abstracts/class-llms-rest-posts-controller.php

```diff
@@ -350,6 +350,35 @@ public function get_item_permissions_check( $request ) {
 		return true;
 	}
 
+	/**
+	 * Retrieves the query params for the objects collection
+	 *
+	 * @since [version]
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+
+		$query_params = parent::get_collection_params();
+		$schema       = $this->get_item_schema();
+
+		if ( isset( $schema['properties']['status'] ) ) {
+			$query_params['status'] = array(
+				'default'           => 'publish',
+				'description'       => __( 'Limit result set to posts assigned one or more statuses.', 'lifterlms' ),
+				'type'              => 'array',
+				'items'             => array(
+					'enum' => array_merge( array_keys( get_post_statuses() ), array( 'future', 'trash', 'auto-draft', 'any' ) ),
```

Comment:
I think that we can be more future proof using `get_post_stati()`, even though it will allow "private" statuses to be used and also statuses that are invalid for the given post type. Although technically I don't think you can register a status to a specific post type. And, in any event, the WP core is using this in their posts controller:

https://core.trac.wordpress.org/browser/tags/5.7/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2768

If someone queries `llms-completed` on a course they'll get an empty list. Life goes on, right? I don't think we expose anything dangerous in doing so.

The same would happen if someone queries the same post status through the WP core rest on regular old post. No results. Life goes on, right?

And in this same way if they add some new post status like "archived" -- or I don't know, anything -- and it isn't included in the hardcoded response of `get_post_statuses()` we won't have to update our endpoint to support it.

Reply:
I think you're right, although I think i'm not very comfortable with this enum "allowing" more than it's actually needed, but ok.

I will also change the schema `status` as we allow 'auto-draft' and 'trash'. That's wrong.
I think I will go with 
`get_post_stati( array( 'internal' => false, '_built_in' => true ) );` for our normal post types, then the orders, transactions and quizzes will override this.

`array_keys( get_post_stati( array( 'internal' => false, '_built_in' => true ) ) );` => 
```
    [0] => publish
    [1] => future
    [2] => draft
    [3] => pending
    [4] => private
```

while currently we have:
`array_merge( array_keys( get_post_statuses() ), array( 'future', 'trash', 'auto-draft' ) )` => 
```
    [0] => draft
    [1] => pending
    [2] => private
    [3] => publish
    [4] => future
    [5] => trash
    [6] => auto-draft
```

Going to push this change, then there's time for your review :P